### ISSUE TITLE
Failed to deserialize immutable Collections.SingletonMap

### ISSUE THREAD

**Issue by PavelVinogradov** [jdereg/json-io#73]
Related to #32 , but a bit different and can't be solved with assignInstantiator.

json-IO fails to deserialize Collections.SingletonMap instance:

```
    @Test
    void testSingletonMap()
    {
        Map root1 = Collections.singletonMap( "testCfgKey", "testCfgValue" )
        String json = JsonWriter.objectToJson(root1)
        Map root2 = (Map) JsonReader.jsonToJava(json)
    }
```

> Caused by: java.lang.UnsupportedOperationException
>   at java.util.AbstractMap.put(AbstractMap.java:209)
>   at com.cedarsoftware.util.io.Resolver.rehashMaps(Resolver.java:556)
>   at com.cedarsoftware.util.io.Resolver.cleanup(Resolver.java:139)
>   at com.cedarsoftware.util.io.JsonReader.convertParsedMapsToJava(JsonReader.java:596)

Discussion about the same issue in Jackson: https://github.com/FasterXML/jackson-core/issues/218

**Comment by zfarrell** [comment]
Any update on this?

**Comment by jdereg** [comment]
This is now implemented and available in 4.11.0.
